### PR TITLE
Symmetrical RSS

### PR DESCRIPTION
**Pull request by WiednerF**
This Pull Request add a custom Hash Key and a Flag setting to be able to use Symmetrical RSS on hardware level (ensuring that TCP bidirectional flows go to the same ending). The Flag is optional and by default flase using the same as it was before. So all prior examples work without changing with the default hardware key, but are able to use the symmetrical hash key as well.

### CODE REVIEW

File: src/device.c [hunk 1]
```diff
@@ -111,8 +115,8 @@ int dpdk_configure_device(struct libmoon_device_config* cfg) {
 	};
 
 	struct rte_eth_rss_conf rss_conf = {
-		.rss_key = NULL,
-		.rss_key_len = 0,
+		.rss_key = cfg->enable_rss_symm?symm_rss_hash_key:NULL,
+		.rss_key_len = cfg->enable_rss_symm?SYMM_RSS_HASHKEY_LENGTH:0,
```
Comment (emmericp) [inline src/device.c]: please use spaces around `foo ? x : bar`

**Comment by emmericp** [comment]
please get rid of the submodule updates (unless there is a good reason to do so)

**Comment by WiednerF** [comment]
Thanks for the Feedback. This was my failure with git. Sorry. Both are resolved now.

**Comment by emmericp** [comment]
I don't understand the new option `includeSharedCores`: why would it depend on the host? maxCores should never be more than the number of cores you have.

Can you give an example where this currently fails?

**Comment by WiednerF** [comment]
Sorry, this should not be part of this pull request. I added it to the false branch. I will start a new Pull Request from a Feature Branch instead.

**Pull request closed without merging**